src/Microsoft.ML.DataView/DataViewType.cs

```diff
@@ -461,4 +480,48 @@ public override bool Equals(DataViewType other)
 
         public override string ToString() => "TimeSpan";
     }
+
+    /// <summary>
+    /// <see cref="DataViewTypeAttribute"/> should be used to decorated class properties and fields, if that class' instances will be loaded as ML.NET <see cref="IDataView"/>.
+    /// The function <see cref="Register"/> will be called to register a <see cref="DataViewType"/> for a <see cref="Type"/> with its <see cref="Attribute"/>s.
+    /// Whenever a value typed to the registered <see cref="Type"/> and its <see cref="Attribute"/>s, its <see cref="DataViewSchema.Column.Type"/> would be the
+    /// associated <see cref="DataViewType"/>.
+    /// </summary>
+    public abstract class DataViewTypeAttribute : Attribute
```

Comment:
This class could implement `IEquatable<DataViewTypeAttribute>`, just like `DataViewType` does.